### PR TITLE
get_session_info

### PR DESCRIPTION
**Pull request by yp2800**
fix undefine colums

### CODE REVIEW

File: racadm.py [hunk 1]
```diff
@@ -133,28 +141,29 @@ def logout(self):
...
         Login to the drac server and get a cookie. Use details pased during initialisation
         '''
-        if 'sid' in self.session.cookies and self.session.cookies['sid'] > 0:
-            self.logger.debug('We already have a session')
-            self.login_state = 'OK'
-            return True
+        # if 'sid' in self.session.cookies and self.session.cookies['sid'] != 0:
+            # self.logger.debug('We already have a session')
+            # self.login_state = 'OK'
+            # return True
```
Comment (b4ldr) [inline racadm.py]: Why did you remove this, this significantly speeds up subsequent calls to the libraryy

File: racadm.py [hunk 2]
```diff
@@ -207,18 +218,18 @@ def _raw_command(self, command):
...
+
         @command = command to run
         @retry = Number of time to retry the login
         '''
         tries=0
         status, message = self._raw_command(command)
         if status == RacStatus.RAC_STATUS_FAILED and tries < retry:
             while tries < retry:
-                self.session.cookies['sid'] = 0
+                self.session.cookies['sid'] = '0'
```
Comment (b4ldr) [inline racadm.py]: Why have you changed this to a string did you have an error

File: racadm.py [hunk 3]
```diff
@@ -661,17 +666,141 @@ def arg_parse():
...
+                except Exception:
+                        print ip, 'fetch info failed ...'
+        json.dump(all_results, open('./getIdracInfo.json', 'w'))
+        return
+
+    if args.hostname == 'localhost':
+        print 'python getIdracInfo.py -H ip -u root -p xxxxxxx || python getIdracInfo.py -f ip.list -u root -p xxxxxx '
+        return
+
+    if check_ping(args.hostname) == 'Network Active':
```
Comment (b4ldr) [inline racadm.py]: connecting the host should not be reliant on icmp, can you please make this optinal.

**Review by b4ldr — COMMENTED** [review]
most changes seem pretty good, could you please address comments and also squash the pull request.  Also sorry its taken a while to comment on this i missed your comments

File: racadm.py (hunk 1 above)
Comment (yp2800) [reply racadm.py]: Session is not valid,when second call class function.
I suspect the login's problem.
 so code annotation, it works.
but why this problem come? the dell idrace server setting's problem?
thank you

File: racadm.py (hunk 2 above)
Comment (yp2800) [reply racadm.py]: Yeah，it's wrong.

File: racadm.py (hunk 3 above)
Comment (yp2800) [reply racadm.py]: yes,it will be block sometimes ,I adjust for my environment. thank you

File: racadm.py [hunk 4]
```diff
@@ -133,71 +137,74 @@ def logout(self):
         Close the session so we are not stealling spaces
         '''
         #this dosn't work because 'Closing the executing session is not permitted.'
```
Comment (jn01674361) [inline racadm.py]: Do you know how to work around the "Closing the executing session is not permitted."-error? I'm writing a script for configuring servers using the iDrac and cannot find anything online as to how to fix this.

Comment (b4ldr) [reply racadm.py]: afraid not i think i generally do a `racreset` once im done.  Also keep in mind if you are using this script that the syntax i use is being deprecated so may not work in future.  unfortunatly i havn't had time to rewrite for the new interface yet.

**Pull request closed without merging**